compiler/src/dotty/tools/dotc/transform/TailRec.scala

```diff
@@ -444,7 +444,7 @@ class TailRec extends MiniPhase {
 
         case Return(expr, from) =>
           val fromSym = from.symbol
-          val inTailPosition = fromSym.is(Label) && tailPositionLabeledSyms.contains(fromSym)
+          val inTailPosition = true
```

Comment:
This change will consider as tail-call a `return@label` where the label is not itself in tail position. This is incorrect. It should be something like
```suggestion
          val inTailPosition = !fromSym.is(Label) || tailPositionLabeledSyms.contains(fromSym)
```

Reply:
Done ✅